### ISSUE TITLE
Overuse of currentUser

### ISSUE THREAD

**Issue by SirPython** [Cardshifter/HTML-Client#57]
In some places, there may be an overuse of currentUser. For example, does the game board *really* need to store the each user's properties there?

**Comment by danpantry** [comment]
Where is game board even getting `currentUser` from? I looked over the source and its not being injected via the constructor nor is it supplied via the template.

Either way, you're right, this smells. That controller is way too big and does too much.

**Comment by SirPython** [comment]
@danpantry I completely agree about the controllers doing too much. Now looking back on it, it's like reading C code where everything is in the main method (I plan on starting to fix this after the client is fully playable).

The `currentUser` comes from [`src/login/controller.js`](https://github.com/Cardshifter/HTML-Client/blob/master/src/login/controller.js#L18).

**Comment by Zomis** [comment]
Seems like game board doesn't use `currentUser` at all anymore (which I am grateful for), I'd like however that it also be removed in https://github.com/Cardshifter/HTML-Client/blob/develop/src/deck_builder/controller.js#L15 as well. Is it possible to pass it along to the controller constructor or something, rather than have the controller ask the `currentUser` object for this value? ("Tell, don't ask" principle)

**Comment by Zomis** [comment]
With VueJS, it's passed around as a property to all components. The game board needs it for game id, mod name and player index. Deck builder needs it for the mod name and to pass it on to game board.
